Allow links to Facebook Messenger

csherland opened the pull request:
Because `amp-share` is still experimental I'd like to allow links using the `fb-messenger` protocol so that it can be used as a share element.

dvoytenko commented:
/to @Gregable @mkhatib

mkhatib commented:
This LGTM, assigning to @powdercloud just because not sure if this will be reflected and deployed or if there are more stuff needed to be done.

Also, @csherland can you whitelist the protocol in the `amp-social-share` validations as well? Even when its out of experimental we'd need to whitelist protocols like we do with anchors.

Gregable commented:
Just to check what you mean, you just want stuff like `<a href="fb-messenger:foo">` right?

powdercloud commented:
@mkhatib is correct:
https://github.com/ampproject/amphtml/blob/master/extensions/amp-social-share/0.1/validator-amp-social-share.protoascii

So this is the thing to change for amp-social-share, and getting this done here in Github is all there is to be done except for us to pull it into google3 (our internal stuff) and release it, which we'll do.

Changing validator-main.protoascii - we've thus far not done this via Github, but rather pushed changes out into Github. So if we want to allow the fb-messenger for the links as well (like, regular a tags), then I'm happy to make the change internally and push it out. You'd want to revert this part of your PR. This is a bit of a quirky process but by making the pulling / pushing a one-way street per file it avoids us having to do nontrivial merges unless they're really unavoidable. My apologies, and thanks much for the PR, @csherland.

csherland commented:
@Gregable: Yes, that's correct

@powdercloud: I think I understood you correctly, but could you please verify? I've removed by changes from `validator/validator-main.protoascii`  and moved the change to `amp-social-share/0.1/validator-amp-social-share.protoascii`

powdercloud commented on extensions/amp-social-share/0.1/validator-amp-social-share.protoascii:
```diff
@@ -64,6 +64,7 @@ tags: {  # <amp-social-share>
       allowed_protocol: "mailto"
       # Whitelisting additional commonly observed third party
       # protocols which should be safe
+      allower_protocol: "fb-messenger"
```
It should be allowed_protocol: (typo).

powdercloud commented:
@csherland Thank you. The change is good in spirit (except for the typo).

csherland commented:
@powdercloud: oops 😅  should be fixed

powdercloud commented:
Thanks much!
